component/libfuse/libfuse_handler.go

```diff
@@ -544,14 +544,15 @@ func libfuse_open(path *C.char, fi *C.fuse_file_info_t) C.int {
 	name = common.NormalizeObjectName(name)
 	log.Trace("Libfuse::libfuse_open : %s", name)
 	// TODO: Should this sit behind a user option? What if we change something to support these in the future?
-	// Mask out SYNC flags since write operation will fail
-	if fi.flags&C.O_SYNC != 0 {
+	// Mask out SYNC and DIRECT flags since write operation will fail
+	if fi.flags&C.O_SYNC != 0 || fi.flags&C.__O_DIRECT != 0 {
 		log.Err("Libfuse::libfuse_open : Reset flags for open %s, fi.flags %X", name, fi.flags)
-		// Blobfuse2 does not support the SYNC flag. If a user application passes this flag on to blobfuse2
+		// Blobfuse2 does not support the SYNC or DIRECT flag. If a user application passes this flag on to blobfuse2
 		// and we open the file with this flag, subsequent write operations wlil fail with "Invalid argument" error.
 		// Mask them out here in the open call so that write works.
 		// Oracle RMAN is one such application that sends these flags during backup
 		fi.flags = fi.flags &^ C.O_SYNC
+		fi.flags = fi.flags &^ C.__O_DIRECT
```

Comment:
Shall we add a UT for this where this flag is given from user end and we just ensure that read/write to file does not fail.